tests/unit/test_flight.py

```diff
@@ -220,6 +222,18 @@ def test_flight_validate(flight_data: pd.DataFrame) -> None:
         Flight(df)
 
 
+def test_flight_fitting() -> None:
+    ds = pd.read_parquet(get_static_path("flight-spire-data-cleaning.pq"))
+    ds.rename(columns={"altitude_baro": "altitude_ft", "timestamp": "time"}, inplace=True)
+    ds["time"] = ds["time"].dt.tz_localize(None)
+    flight = Flight(ds[ds["callsign"] == "BAW506"], drop_duplicated_times=True)
+    smoothed_flight = flight.fit_flight_profile()
```

Comment:
Could you add an assert here that the original `flight` isn't modified by this? IDK the best way to do this -- you could pin the mean before you call the `fit_flight_profile` method, and check that after you call it, the mean hasn't changed.